### PR TITLE
Detect and error on conflicting routes between plugins

### PR DESCRIPTION
**Pull request by mosch**
## Summary

Add route conflict detection to prevent multiple plugins from silently shadowing each other when they register the same path. In development mode, the app now throws a `ZudokuError` if conflicts are detected, with a helpful error message and developer hint.

## Changes

- **New module `detectRouteConflicts.ts`**: Implements conflict detection by collecting all absolute paths from plugin route trees, accounting for:
  - Relative vs. absolute child paths (relative paths are joined with parent, absolute paths are kept as-is)
  - Trailing slash normalization via `joinUrl()`
  - Deduplication within a single plugin (legitimate parent/child path repetition)
  - Pathless layout routes (ignored)

- **New test suite `detectRouteConflicts.test.ts`**: Comprehensive test coverage for all edge cases including disjoint routes, duplicate paths, path normalization, nested routes, and sorting

- **Updated `main.tsx`**: 
  - Refactored route collection to preserve per-plugin route arrays for conflict checking
  - Added dev-mode conflict detection that throws a `ZudokuError` with actionable guidance
  - Removed TODO comment about route conflict detection

## Implementation Details

The conflict detection runs only in development mode (`import.meta.env.DEV`) to avoid runtime overhead in production. When conflicts are found, the error includes:
- A clear message listing all conflicting paths
- A developer hint suggesting to check for overlapping `path`s or `basePath`s in plugins
- Proper pluralization of "route" vs "routes"

The `collectPaths()` function recursively traverses route trees and handles the nuance that zudoku plugins author child paths as absolute (already basePath-joined), so only relative child paths are joined with their parent.

https://claude.ai/code/session_01RGUndHnRZeugHft6LLJjvS

### CODE REVIEW

File: packages/zudoku/src/app/detectRouteConflicts.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import type { RouteObject } from "react-router";
+import { joinUrl } from "../lib/util/joinUrl.js";
+
+// Collect the absolute path of every route that declares one. Child paths in
+// zudoku plugins are authored as absolute (already basePath-joined), so only
+// join with the parent when the child path is relative.
```
Comment (dan-lee) [inline packages/zudoku/src/app/detectRouteConflicts.ts]: Is this valid? I thought zudoku always provides full paths or am I mistaken?

File: packages/zudoku/src/app/detectRouteConflicts.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  for (const route of routes) {
+    const fullPath =
+      route.path != null
+        ? route.path.startsWith("/")
+          ? joinUrl(route.path)
+          : joinUrl(parent, route.path)
+        : parent;
+
+    if (route.path != null) acc.add(fullPath);
+    if (route.children) collectPaths(route.children, fullPath, acc);
```
[the remark this answers is not in the document]
Comment (mosch) [reply packages/zudoku/src/app/detectRouteConflicts.ts:24]: Good catch — fixed in 038b496. `collectPaths()` now records index routes (`{ index: true }` without a `path`) against their parent path, or `/` at the top level.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/zudoku/src/app/detectRouteConflicts.test.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    expect(detectRouteConflicts([a, b])).toEqual(["/api/tags"]);
+  });
+
+  it("reports every conflicting path sorted", () => {
+    const a: RouteObject[] = [{ path: "/b" }, { path: "/a" }];
+    const b: RouteObject[] = [{ path: "/a" }, { path: "/b" }];
+
+    expect(detectRouteConflicts([a, b])).toEqual(["/a", "/b"]);
+  });
```
[the remark this answers is not in the document]
Comment (mosch) [reply packages/zudoku/src/app/detectRouteConflicts.test.ts:63]: Added in 038b496 — tests now cover a top-level index route conflicting with an explicit `/` and a nested index route conflicting with its parent path.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by dan-lee — APPROVED** [review]
